This patch adds support for the GTA01Bv3 revision of the FIC Neo1973 smartphone.

GTA01Bv3 is the "Phase 0" revision (due to be shipped on Feb 11, 2007).

Signed-off-by: Harald Welte <laforge@openmoko.org>

Index: u-boot.git/Makefile
===================================================================
--- u-boot.git.orig/Makefile	2007-02-06 14:11:13.000000000 +0100
+++ u-boot.git/Makefile	2007-02-06 14:11:25.000000000 +0100
@@ -1910,6 +1910,7 @@
 gta01_config \
 gta01v3_config \
 gta01bv2_config \
+gta01bv3_config \
 gta01v4_config :	unconfig
 	@board/neo1973/split_by_variant.sh $@
 
Index: u-boot.git/board/neo1973/lowlevel_init.S
===================================================================
--- u-boot.git.orig/board/neo1973/lowlevel_init.S	2007-02-06 14:13:28.000000000 +0100
+++ u-boot.git/board/neo1973/lowlevel_init.S	2007-02-06 14:13:51.000000000 +0100
@@ -111,7 +111,7 @@
 #define B6_Trcd	 	 	0x1	/* 3clk */
 #if defined (CONFIG_ARCH_GTA01_v3) || defined(CONFIG_ARCH_GTA01_v4)
 #define B6_SCAN		 	0x1	/* 9bit */
-#elif defined(CONFIG_ARCH_GTA01B_v2)
+#elif defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 #define B6_SCAN		 	0x2	/* 10bit */
 #endif
 
@@ -153,7 +153,7 @@
 	orr	r1, r1, #0xc0000000
 	mcr	p15, 0, r1, c1, c0, 0
 
-#if defined(CONFIG_ARCH_GTA01_v4) || defined(CONFIG_ARCH_GTA01B_v2)
+#if defined(CONFIG_ARCH_GTA01_v4) || defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 	/* switch on power for NAND */
 	ldr	r0, =0x56000010	/* GPBCON */
 	ldr	r1, [r0]
Index: u-boot.git/board/neo1973/neo1973.c
===================================================================
--- u-boot.git.orig/board/neo1973/neo1973.c	2007-02-06 14:09:44.000000000 +0100
+++ u-boot.git/board/neo1973/neo1973.c	2007-02-06 14:11:04.000000000 +0100
@@ -43,7 +43,7 @@
 #define M_MDIV	0x90		/* Fout = 202.8MHz */
 #define M_PDIV	0x7
 #define M_SDIV	0x0
-#elif defined(CONFIG_ARCH_GTA01B_v2)
+#elif defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 #if 0
 #define M_MDIV	0x7d		/* Fout = 266MHz */
 #define M_PDIV	0x1
@@ -140,7 +140,7 @@
 
 	gpio->GPHCON = 0x0000FAAA;
 	gpio->GPHUP = 0x000007FF;
-#elif defined(CONFIG_ARCH_GTA01B_v2)
+#elif defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 	gpio->GPACON = 0x005E47FF;
 
 	gpio->GPBCON = 0x00145416;
@@ -194,7 +194,7 @@
 	tmp = 0x0f; /* CHGAPE | AUTOFST | fast charge */
 	i2c_write(0x08, 0x29, 1, &tmp, 1);
 
-#if defined(CONFIG_ARCH_GTA01B_v2)
+#if defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 	tmp = 0xe4;	/* CPU Voltage: 2.1V */
 	i2c_write(0x08, 0x21, 1, &tmp, 1);
 #endif
@@ -223,5 +223,7 @@
 	return 0x00000140;
 #elif defined(CONFIG_ARCH_GTA01B_v2)
 	return 0x00000220;
+#elif defined(CONFIG_ARCH_GTA01B_v3)
+	return 0x00000230;
 #endif
 }
Index: u-boot.git/board/neo1973/split_by_variant.sh
===================================================================
--- u-boot.git.orig/board/neo1973/split_by_variant.sh	2007-02-06 14:11:29.000000000 +0100
+++ u-boot.git/board/neo1973/split_by_variant.sh	2007-02-06 14:12:07.000000000 +0100
@@ -9,8 +9,9 @@
 mkdir -p ${obj}include
 if [ "$1" == "" ]
 then
-	echo "$0:: No parameters - using GTA01v3 config"
-	echo "#define CONFIG_ARCH_GTA01_v3" > ${obj}include/config.h
+	echo "$0:: No parameters - using GTA01Bv3 config"
+	echo "#define CONFIG_ARCH_GTA01B_v3" > ${obj}include/config.h
+	echo "GTA01_BIG_RAM=y" > ${obj}board/neo1973/config.tmp
 else
 	case "$1" in
 	gta01v4_config)
@@ -28,6 +29,11 @@
 	echo "GTA01_BIG_RAM=y" > ${obj}board/neo1973/config.tmp
 	;;
 
+	gta01bv3_config)
+	echo "#define CONFIG_ARCH_GTA01B_v3" > ${obj}include/config.h
+	echo "GTA01_BIG_RAM=y" > ${obj}board/neo1973/config.tmp
+	;;
+
 	*)
 	echo "$0:: Unrecognised config - using gta01v3_config"
 	echo "#define CONFIG_ARCH_GTA01_v3" > ${obj}include/config.h
Index: u-boot.git/drivers/usbdcore_s3c2410.c
===================================================================
--- u-boot.git.orig/drivers/usbdcore_s3c2410.c	2007-02-06 14:14:44.000000000 +0100
+++ u-boot.git/drivers/usbdcore_s3c2410.c	2007-02-06 14:15:15.000000000 +0100
@@ -595,7 +595,7 @@
 void udc_connect (void)
 {
 	UDCDBG ("connect, enable Pullup");
-#if defined(CONFIG_ARCH_GTA01_v4) || defined(CONFIG_ARCH_GTA01B_v2)
+#if defined(CONFIG_ARCH_GTA01_v4) || defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 	S3C24X0_GPIO * const gpio = S3C24X0_GetBase_GPIO();
 
 	/* create a short disconnect, since we might come out of reset */
@@ -613,7 +613,7 @@
 {
 	UDCDBG ("disconnect, disable Pullup");
 	S3C24X0_INTERRUPT * irq = S3C24X0_GetBase_INTERRUPT();
-#if defined(CONFIG_ARCH_GTA01_v4) || defined(CONFIG_ARCH_GTA01B_v2)
+#if defined(CONFIG_ARCH_GTA01_v4) || defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 	S3C24X0_GPIO * const gpio = S3C24X0_GetBase_GPIO();
 
 	//s3c2410_gpio_setpin(GTA01_GPIO_USB_PULLUP, 0);
Index: u-boot.git/include/configs/neo1973.h
===================================================================
--- u-boot.git.orig/include/configs/neo1973.h	2007-02-06 14:14:01.000000000 +0100
+++ u-boot.git/include/configs/neo1973.h	2007-02-06 14:14:25.000000000 +0100
@@ -138,6 +138,8 @@
 #define	CFG_PROMPT		"GTA01v4 # "	/* Monitor Command Prompt	*/
 #elif defined(CONFIG_ARCH_GTA01B_v2)
 #define	CFG_PROMPT		"GTA01Bv2 # "	/* Monitor Command Prompt	*/
+#elif defined(CONFIG_ARCH_GTA01B_v3)
+#define	CFG_PROMPT		"GTA01Bv3 # "	/* Monitor Command Prompt	*/
 #endif
 #define	CFG_CBSIZE		256		/* Console I/O Buffer Size	*/
 #define	CFG_PBSIZE (CFG_CBSIZE+sizeof(CFG_PROMPT)+16) /* Print Buffer Size */
@@ -192,7 +194,7 @@
 #define PHYS_SDRAM_1		0x30000000 /* SDRAM Bank #1 */
 #if defined(CONFIG_ARCH_GTA01_v3) || defined(CONFIG_ARCH_GTA01_v4)
 #define PHYS_SDRAM_1_SIZE	0x04000000 /* 64 MB */
-#elif defined(CONFIG_ARCH_GTA01B_v2)
+#elif defined(CONFIG_ARCH_GTA01B_v2) || defined(CONFIG_ARCH_GTA01B_v3)
 #define PHYS_SDRAM_1_SIZE	0x08000000 /* 128 MB */
 #else
 #error Please define GTA01 variant
